code/game/machinery/candymachine.dm

```diff
@@ -0,0 +1,25 @@
+//Candy Machine shamelessly created by ripping apart the gashapon machine and frankensteining together this horrifying new creation.
+
+
+/obj/machinery/sweet
+	name = "Sweet Machine"
+	desc = "Insert coin, recieve a sweet!"
+	icon = 'icons/obj/candymachine.dmi'
+	icon_state = "sweetmachine"
+	anchored = 1
+	density = 1
+	machine_flags = WRENCHMOVE | FIXED2WORK
+
+/obj/machinery/sweet/attackby(var/obj/O as obj, var/mob/user as mob)
+	if (istype(O, /obj/item/weapon/coin/))
```

Comment:
Could probably use a check for `stat & (NOPOWER | BROKEN)`, since even though you have `FIXED2WORK`, it'll still work work.